{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7c3a7a00",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: selenium in c:\\programdata\\anaconda3\\lib\\site-packages (4.1.3)\n",
      "Requirement already satisfied: trio-websocket~=0.9 in c:\\programdata\\anaconda3\\lib\\site-packages (from selenium) (0.9.2)\n",
      "Requirement already satisfied: trio~=0.17 in c:\\programdata\\anaconda3\\lib\\site-packages (from selenium) (0.20.0)\n",
      "Requirement already satisfied: urllib3[secure,socks]~=1.26 in c:\\programdata\\anaconda3\\lib\\site-packages (from selenium) (1.26.7)\n",
      "Requirement already satisfied: cffi>=1.14 in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (1.14.6)\n",
      "Requirement already satisfied: idna in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (3.2)\n",
      "Requirement already satisfied: outcome in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (1.1.0)\n",
      "Requirement already satisfied: sortedcontainers in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (2.4.0)\n",
      "Requirement already satisfied: async-generator>=1.9 in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (1.10)\n",
      "Requirement already satisfied: sniffio in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (1.2.0)\n",
      "Requirement already satisfied: attrs>=19.2.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from trio~=0.17->selenium) (21.2.0)\n",
      "Requirement already satisfied: pycparser in c:\\programdata\\anaconda3\\lib\\site-packages (from cffi>=1.14->trio~=0.17->selenium) (2.20)\n",
      "Requirement already satisfied: wsproto>=0.14 in c:\\programdata\\anaconda3\\lib\\site-packages (from trio-websocket~=0.9->selenium) (1.1.0)\n",
      "Requirement already satisfied: PySocks!=1.5.7,<2.0,>=1.5.6 in c:\\programdata\\anaconda3\\lib\\site-packages (from urllib3[secure,socks]~=1.26->selenium) (1.7.1)\n",
      "Requirement already satisfied: pyOpenSSL>=0.14 in c:\\programdata\\anaconda3\\lib\\site-packages (from urllib3[secure,socks]~=1.26->selenium) (21.0.0)\n",
      "Requirement already satisfied: cryptography>=1.3.4 in c:\\programdata\\anaconda3\\lib\\site-packages (from urllib3[secure,socks]~=1.26->selenium) (3.4.8)\n",
      "Requirement already satisfied: certifi in c:\\programdata\\anaconda3\\lib\\site-packages (from urllib3[secure,socks]~=1.26->selenium) (2021.10.8)\n",
      "Requirement already satisfied: six>=1.5.2 in c:\\programdata\\anaconda3\\lib\\site-packages (from pyOpenSSL>=0.14->urllib3[secure,socks]~=1.26->selenium) (1.16.0)\n",
      "Requirement already satisfied: h11<1,>=0.9.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from wsproto>=0.14->trio-websocket~=0.9->selenium) (0.13.0)\n"
     ]
    }
   ],
   "source": [
    "! pip install selenium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4d637886",
   "metadata": {},
   "outputs": [],
   "source": [
    "from selenium import webdriver\n",
    "from selenium.webdriver.common.by import By\n",
    "from selenium.webdriver import Keys\n",
    "from selenium.webdriver.common.action_chains import ActionChains\n",
    "from time import sleep\n",
    "import random\n",
    "import json\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5cf2c97a",
   "metadata": {},
   "outputs": [],
   "source": [
    "search_name = \"Духи\"\n",
    "articul_num = \"1713520\"\n",
    "post_place = \"г Москва, Улица Лазо 4к1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "39c17991",
   "metadata": {},
   "outputs": [],
   "source": [
    "_bot = bot\n",
    "bot = Bot(driver=_bot.driver)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "834c5694",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Python Modules\n",
    "import os\n",
    "import json\n",
    "import logging\n",
    "import pandas as pd\n",
    "\n",
    "# For telegram api\n",
    "# pip install python-telegram-bot --upgrade\n",
    "from telegram import Update, InlineKeyboardButton, InlineKeyboardMarkup\n",
    "from telegram.ext import Updater, CommandHandler, MessageHandler, Filters, CallbackContext, \\\n",
    "    CallbackQueryHandler, TypeHandler, ConversationHandler\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c3c54120",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "run bot buy\n",
      "price 1522\n",
      "price_filter started\n",
      "min_price 1100 max_price 1900\n",
      "price_filter ended\n",
      "['44398622']\n",
      "['44398622']\n"
     ]
    },
    {
     "ename": "WebDriverException",
     "evalue": "Message: chrome not reachable\n  (Session info: chrome=100.0.4896.88)\nStacktrace:\nBacktrace:\n\tOrdinal0 [0x00679943+2595139]\n\tOrdinal0 [0x0060C9F1+2148849]\n\tOrdinal0 [0x005043F0+1065968]\n\tOrdinal0 [0x004F87C2+1017794]\n\tOrdinal0 [0x004F8FF8+1019896]\n\tOrdinal0 [0x004FA892+1026194]\n\tOrdinal0 [0x004F4219+999961]\n\tOrdinal0 [0x00505860+1071200]\n\tOrdinal0 [0x0055B2D2+1422034]\n\tOrdinal0 [0x0054B806+1357830]\n\tOrdinal0 [0x00526086+1204358]\n\tOrdinal0 [0x00526F96+1208214]\n\tGetHandleVerifier [0x0081B232+1658114]\n\tGetHandleVerifier [0x008D312C+2411516]\n\tGetHandleVerifier [0x0070F261+560433]\n\tGetHandleVerifier [0x0070E366+556598]\n\tOrdinal0 [0x0061286B+2173035]\n\tOrdinal0 [0x006175F8+2192888]\n\tOrdinal0 [0x006176E5+2193125]\n\tOrdinal0 [0x006211FC+2232828]\n\tBaseThreadInitThunk [0x7683FA29+25]\n\tRtlGetAppContainerNamedObjectPath [0x77207A7E+286]\n\tRtlGetAppContainerNamedObjectPath [0x77207A4E+238]\n",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mKeyboardInterrupt\u001B[0m                         Traceback (most recent call last)",
      "\u001B[1;32m~\\WBbot\\Basket.py\u001B[0m in \u001B[0;36mget_qr_code\u001B[1;34m(self, order_id, bot_name)\u001B[0m\n\u001B[0;32m    107\u001B[0m         \u001B[1;32mtry\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 108\u001B[1;33m             self.driver.find_element(By.XPATH,\n\u001B[0m\u001B[0;32m    109\u001B[0m                                      '//button[text()=\"                Оплатить заказ                \"]').click()\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001B[0m in \u001B[0;36mfind_element\u001B[1;34m(self, by, value)\u001B[0m\n\u001B[0;32m   1243\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m-> 1244\u001B[1;33m         return self.execute(Command.FIND_ELEMENT, {\n\u001B[0m\u001B[0;32m   1245\u001B[0m             \u001B[1;34m'using'\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0mby\u001B[0m\u001B[1;33m,\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001B[0m in \u001B[0;36mexecute\u001B[1;34m(self, driver_command, params)\u001B[0m\n\u001B[0;32m    421\u001B[0m         \u001B[0mparams\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0m_wrap_value\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mparams\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 422\u001B[1;33m         \u001B[0mresponse\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mcommand_executor\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mexecute\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mdriver_command\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mparams\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    423\u001B[0m         \u001B[1;32mif\u001B[0m \u001B[0mresponse\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\remote_connection.py\u001B[0m in \u001B[0;36mexecute\u001B[1;34m(self, command, params)\u001B[0m\n\u001B[0;32m    420\u001B[0m         \u001B[0murl\u001B[0m \u001B[1;33m=\u001B[0m \u001B[1;34mf\"{self._url}{path}\"\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 421\u001B[1;33m         \u001B[1;32mreturn\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0m_request\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mcommand_info\u001B[0m\u001B[1;33m[\u001B[0m\u001B[1;36m0\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0murl\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mbody\u001B[0m\u001B[1;33m=\u001B[0m\u001B[0mdata\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    422\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\remote_connection.py\u001B[0m in \u001B[0;36m_request\u001B[1;34m(self, method, url, body)\u001B[0m\n\u001B[0;32m    442\u001B[0m         \u001B[1;32mif\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mkeep_alive\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 443\u001B[1;33m             \u001B[0mresp\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0m_conn\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mrequest\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmethod\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0murl\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mbody\u001B[0m\u001B[1;33m=\u001B[0m\u001B[0mbody\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mheaders\u001B[0m\u001B[1;33m=\u001B[0m\u001B[0mheaders\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    444\u001B[0m             \u001B[0mstatuscode\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mresp\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mstatus\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\request.py\u001B[0m in \u001B[0;36mrequest\u001B[1;34m(self, method, url, fields, headers, **urlopen_kw)\u001B[0m\n\u001B[0;32m     77\u001B[0m         \u001B[1;32melse\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m---> 78\u001B[1;33m             return self.request_encode_body(\n\u001B[0m\u001B[0;32m     79\u001B[0m                 \u001B[0mmethod\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0murl\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mfields\u001B[0m\u001B[1;33m=\u001B[0m\u001B[0mfields\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mheaders\u001B[0m\u001B[1;33m=\u001B[0m\u001B[0mheaders\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;33m**\u001B[0m\u001B[0murlopen_kw\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\request.py\u001B[0m in \u001B[0;36mrequest_encode_body\u001B[1;34m(self, method, url, fields, headers, encode_multipart, multipart_boundary, **urlopen_kw)\u001B[0m\n\u001B[0;32m    169\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 170\u001B[1;33m         \u001B[1;32mreturn\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0murlopen\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmethod\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0murl\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;33m**\u001B[0m\u001B[0mextra_kw\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\poolmanager.py\u001B[0m in \u001B[0;36murlopen\u001B[1;34m(self, method, url, redirect, **kw)\u001B[0m\n\u001B[0;32m    374\u001B[0m         \u001B[1;32melse\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 375\u001B[1;33m             \u001B[0mresponse\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mconn\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0murlopen\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmethod\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mu\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mrequest_uri\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;33m**\u001B[0m\u001B[0mkw\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    376\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001B[0m in \u001B[0;36murlopen\u001B[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001B[0m\n\u001B[0;32m    698\u001B[0m             \u001B[1;31m# Make the request on the httplib connection object.\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 699\u001B[1;33m             httplib_response = self._make_request(\n\u001B[0m\u001B[0;32m    700\u001B[0m                 \u001B[0mconn\u001B[0m\u001B[1;33m,\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001B[0m in \u001B[0;36m_make_request\u001B[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001B[0m\n\u001B[0;32m    444\u001B[0m                     \u001B[1;31m# Otherwise it looks like a bug in the code.\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 445\u001B[1;33m                     \u001B[0msix\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mraise_from\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0me\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;32mNone\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    446\u001B[0m         \u001B[1;32mexcept\u001B[0m \u001B[1;33m(\u001B[0m\u001B[0mSocketTimeout\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mBaseSSLError\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mSocketError\u001B[0m\u001B[1;33m)\u001B[0m \u001B[1;32mas\u001B[0m \u001B[0me\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\packages\\six.py\u001B[0m in \u001B[0;36mraise_from\u001B[1;34m(value, from_value)\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\urllib3\\connectionpool.py\u001B[0m in \u001B[0;36m_make_request\u001B[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001B[0m\n\u001B[0;32m    439\u001B[0m                 \u001B[1;32mtry\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 440\u001B[1;33m                     \u001B[0mhttplib_response\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mconn\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mgetresponse\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    441\u001B[0m                 \u001B[1;32mexcept\u001B[0m \u001B[0mBaseException\u001B[0m \u001B[1;32mas\u001B[0m \u001B[0me\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\http\\client.py\u001B[0m in \u001B[0;36mgetresponse\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m   1370\u001B[0m             \u001B[1;32mtry\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m-> 1371\u001B[1;33m                 \u001B[0mresponse\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mbegin\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m   1372\u001B[0m             \u001B[1;32mexcept\u001B[0m \u001B[0mConnectionError\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\http\\client.py\u001B[0m in \u001B[0;36mbegin\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m    318\u001B[0m         \u001B[1;32mwhile\u001B[0m \u001B[1;32mTrue\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 319\u001B[1;33m             \u001B[0mversion\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mstatus\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mreason\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0m_read_status\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    320\u001B[0m             \u001B[1;32mif\u001B[0m \u001B[0mstatus\u001B[0m \u001B[1;33m!=\u001B[0m \u001B[0mCONTINUE\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\http\\client.py\u001B[0m in \u001B[0;36m_read_status\u001B[1;34m(self)\u001B[0m\n\u001B[0;32m    279\u001B[0m     \u001B[1;32mdef\u001B[0m \u001B[0m_read_status\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mself\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 280\u001B[1;33m         \u001B[0mline\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mstr\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mfp\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mreadline\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0m_MAXLINE\u001B[0m \u001B[1;33m+\u001B[0m \u001B[1;36m1\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;34m\"iso-8859-1\"\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    281\u001B[0m         \u001B[1;32mif\u001B[0m \u001B[0mlen\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mline\u001B[0m\u001B[1;33m)\u001B[0m \u001B[1;33m>\u001B[0m \u001B[0m_MAXLINE\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\socket.py\u001B[0m in \u001B[0;36mreadinto\u001B[1;34m(self, b)\u001B[0m\n\u001B[0;32m    703\u001B[0m             \u001B[1;32mtry\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 704\u001B[1;33m                 \u001B[1;32mreturn\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0m_sock\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mrecv_into\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mb\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    705\u001B[0m             \u001B[1;32mexcept\u001B[0m \u001B[0mtimeout\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;31mKeyboardInterrupt\u001B[0m: ",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001B[1;31mWebDriverException\u001B[0m                        Traceback (most recent call last)",
      "\u001B[1;32mC:\\Users\\ASTROC~1\\AppData\\Local\\Temp/ipykernel_8112/4066478754.py\u001B[0m in \u001B[0;36m<module>\u001B[1;34m\u001B[0m\n\u001B[0;32m     35\u001B[0m \u001B[1;32mfor\u001B[0m \u001B[0mi\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mbot\u001B[0m \u001B[1;32min\u001B[0m \u001B[0menumerate\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mbots\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     36\u001B[0m     \u001B[0mprint\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;34m'run bot buy'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m---> 37\u001B[1;33m     \u001B[0mreport\u001B[0m \u001B[1;33m+=\u001B[0m \u001B[0mbot\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mbuy\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mdata_for_bots\u001B[0m\u001B[1;33m[\u001B[0m\u001B[0mi\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mpost_places\u001B[0m\u001B[1;33m[\u001B[0m\u001B[0mi\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0morder_id\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m",
      "\u001B[1;32m~\\WBbot\\Bot.py\u001B[0m in \u001B[0;36mbuy\u001B[1;34m(self, data, post_place, order_id)\u001B[0m\n\u001B[0;32m     41\u001B[0m         \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mbasket\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mchoose_post_place\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mpost_place\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     42\u001B[0m         \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mbasket\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mchoose_payment_method\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m---> 43\u001B[1;33m         \u001B[0mreport\u001B[0m\u001B[1;33m[\u001B[0m\u001B[1;34m'qr_code'\u001B[0m\u001B[1;33m]\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mbasket\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mget_qr_code\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0morder_id\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mname\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m     44\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     45\u001B[0m         \u001B[1;32mreturn\u001B[0m \u001B[1;33m[\u001B[0m\u001B[0mreport\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32m~\\WBbot\\Basket.py\u001B[0m in \u001B[0;36mget_qr_code\u001B[1;34m(self, order_id, bot_name)\u001B[0m\n\u001B[0;32m    118\u001B[0m             \u001B[0mfile\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mopen\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;34m\"config/config.config\"\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mread\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    119\u001B[0m             \u001B[0mconfig\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0meval\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mfile\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 120\u001B[1;33m             \u001B[0mUtils\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mlogin\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mdriver\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mconfig\u001B[0m\u001B[1;33m[\u001B[0m\u001B[1;34m'bots'\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m[\u001B[0m\u001B[0mbot_name\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m[\u001B[0m\u001B[1;34m'number'\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    121\u001B[0m             \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mget_qr_code\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0morder_id\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mbot_name\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    122\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32m~\\WBbot\\Utils.py\u001B[0m in \u001B[0;36mlogin\u001B[1;34m(driver, number)\u001B[0m\n\u001B[0;32m     17\u001B[0m     \u001B[1;32mdef\u001B[0m \u001B[0mlogin\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mdriver\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mnumber\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     18\u001B[0m         \u001B[1;31m# vars # 1\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m---> 19\u001B[1;33m         \u001B[0mto_login_btn\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mdriver\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mfind_element\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mBy\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mCLASS_NAME\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;34m'j-main-login'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m     20\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     21\u001B[0m         \u001B[1;31m# resolve # 1\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001B[0m in \u001B[0;36mfind_element\u001B[1;34m(self, by, value)\u001B[0m\n\u001B[0;32m   1242\u001B[0m             \u001B[0mvalue\u001B[0m \u001B[1;33m=\u001B[0m \u001B[1;34m'[name=\"%s\"]'\u001B[0m \u001B[1;33m%\u001B[0m \u001B[0mvalue\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m   1243\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m-> 1244\u001B[1;33m         return self.execute(Command.FIND_ELEMENT, {\n\u001B[0m\u001B[0;32m   1245\u001B[0m             \u001B[1;34m'using'\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0mby\u001B[0m\u001B[1;33m,\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m   1246\u001B[0m             'value': value})['value']\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001B[0m in \u001B[0;36mexecute\u001B[1;34m(self, driver_command, params)\u001B[0m\n\u001B[0;32m    422\u001B[0m         \u001B[0mresponse\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mcommand_executor\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mexecute\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mdriver_command\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mparams\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    423\u001B[0m         \u001B[1;32mif\u001B[0m \u001B[0mresponse\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 424\u001B[1;33m             \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0merror_handler\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mcheck_response\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mresponse\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    425\u001B[0m             response['value'] = self._unwrap_value(\n\u001B[0;32m    426\u001B[0m                 response.get('value', None))\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py\u001B[0m in \u001B[0;36mcheck_response\u001B[1;34m(self, response)\u001B[0m\n\u001B[0;32m    245\u001B[0m                 \u001B[0malert_text\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mvalue\u001B[0m\u001B[1;33m[\u001B[0m\u001B[1;34m'alert'\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mget\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;34m'text'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    246\u001B[0m             \u001B[1;32mraise\u001B[0m \u001B[0mexception_class\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmessage\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mscreen\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mstacktrace\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0malert_text\u001B[0m\u001B[1;33m)\u001B[0m  \u001B[1;31m# type: ignore[call-arg]  # mypy is not smart enough here\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 247\u001B[1;33m         \u001B[1;32mraise\u001B[0m \u001B[0mexception_class\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmessage\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mscreen\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mstacktrace\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    248\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    249\u001B[0m     \u001B[1;32mdef\u001B[0m \u001B[0m_value_or_default\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mself\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mobj\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0mMapping\u001B[0m\u001B[1;33m[\u001B[0m\u001B[0m_KT\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0m_VT\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mkey\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0m_KT\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mdefault\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0m_VT\u001B[0m\u001B[1;33m)\u001B[0m \u001B[1;33m->\u001B[0m \u001B[0m_VT\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;31mWebDriverException\u001B[0m: Message: chrome not reachable\n  (Session info: chrome=100.0.4896.88)\nStacktrace:\nBacktrace:\n\tOrdinal0 [0x00679943+2595139]\n\tOrdinal0 [0x0060C9F1+2148849]\n\tOrdinal0 [0x005043F0+1065968]\n\tOrdinal0 [0x004F87C2+1017794]\n\tOrdinal0 [0x004F8FF8+1019896]\n\tOrdinal0 [0x004FA892+1026194]\n\tOrdinal0 [0x004F4219+999961]\n\tOrdinal0 [0x00505860+1071200]\n\tOrdinal0 [0x0055B2D2+1422034]\n\tOrdinal0 [0x0054B806+1357830]\n\tOrdinal0 [0x00526086+1204358]\n\tOrdinal0 [0x00526F96+1208214]\n\tGetHandleVerifier [0x0081B232+1658114]\n\tGetHandleVerifier [0x008D312C+2411516]\n\tGetHandleVerifier [0x0070F261+560433]\n\tGetHandleVerifier [0x0070E366+556598]\n\tOrdinal0 [0x0061286B+2173035]\n\tOrdinal0 [0x006175F8+2192888]\n\tOrdinal0 [0x006176E5+2193125]\n\tOrdinal0 [0x006211FC+2232828]\n\tBaseThreadInitThunk [0x7683FA29+25]\n\tRtlGetAppContainerNamedObjectPath [0x77207A7E+286]\n\tRtlGetAppContainerNamedObjectPath [0x77207A4E+238]\n"
     ]
    }
   ],
   "source": [
    "%run Bot.py\n",
    "\n",
    "BOTS_NAME = ['Oleg', 'Einstein', 'Boulevard Depo']\n",
    "\n",
    "file_path = \"users_data/794329884_order.xlsx\"\n",
    "inside=True\n",
    "df = pd.read_excel(file_path)\n",
    "\n",
    "orders = [row.tolist() for i, row in df.iterrows()]\n",
    "\n",
    "articles = [order[0] for order in orders]\n",
    "watch_bot = Bot(name=\"Watcher\")\n",
    "for i, article in enumerate(articles):\n",
    "    orders[i] += [watch_bot.get_data_cart(article)]\n",
    "\n",
    "max_bots = max([order[3] for order in orders])\n",
    "bots = [Bot(name=BOTS_NAME[i]) for i in range(max_bots)]\n",
    "data_for_bots = [[] for _ in range(max_bots)]\n",
    "\n",
    "for _, order in enumerate(orders):\n",
    "    if inside:\n",
    "        article, search_key, quantity, pvz_cnt, _, __, additional_data = order\n",
    "    else:\n",
    "        article, search_key, quantity, pvz_cnt, additional_data = order\n",
    "\n",
    "    for i in range(max_bots):\n",
    "        if pvz_cnt > 0:\n",
    "            data_for_bots[i] += [[article, search_key, quantity, additional_data]]\n",
    "        pvz_cnt -= 1\n",
    "\n",
    "post_places = [order[5] for order in orders][:max_bots]\n",
    "order_id = str(orders[0][4])\n",
    "\n",
    "report = []\n",
    "for i, bot in enumerate(bots):\n",
    "    print('run bot buy')\n",
    "    report += bot.buy(data_for_bots[i], post_places[i], order_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "270a60be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "price_filter started\n",
      "1141 1902\n",
      "min_price_field =  <selenium.webdriver.remote.webelement.WebElement (session=\"4cf82b9719ef4987aa066532788b8c3d\", element=\"6b57a28b-6676-4fa1-a6fc-9c15762332af\")>\n",
      "price_filter ended\n"
     ]
    }
   ],
   "source": [
    "min_price, max_price = 1141, 1902\n",
    "\n",
    "print(\"price_filter started\")\n",
    "print(\"min_price\", min_price, \"max_price\", max_price)\n",
    "min_price_field = bot.catalog.min_price_field()\n",
    "min_price_field.click()\n",
    "min_price_field.send_keys(Keys.CONTROL + \"a\")\n",
    "min_price_field.send_keys(Keys.DELETE)\n",
    "min_price_field.send_keys(str(min_price))\n",
    "\n",
    "sleep(1)\n",
    "\n",
    "max_price_field = bot.catalog.max_price_field()\n",
    "max_price_field.click()\n",
    "\n",
    "sleep(1)\n",
    "max_price_field = bot.catalog.max_price_field()\n",
    "max_price_field.click()\n",
    "max_price_field.click()\n",
    "max_price_field.send_keys(Keys.CONTROL + \"a\")\n",
    "max_price_field.send_keys(Keys.DELETE)\n",
    "max_price_field.send_keys(str(max_price))\n",
    "max_price_field.send_keys(Keys.ENTER)\n",
    "print(\"price_filter ended\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "674d1510",
   "metadata": {},
   "outputs": [],
   "source": [
    "def choose_post_place(adress):\n",
    "    bot.driver.find_element(By.XPATH, '//h2[text()=\"Способ доставки\"]/../../div[text()=\"Выбрать адрес доставки\"]').click()\n",
    "    sleep(1)\n",
    "    bot.driver.find_element(By.XPATH, '//button[text()=\"Выбрать адрес доставки\"]').click()\n",
    "    sleep(2)\n",
    "    bot.driver.find_element(By.XPATH, '//input[@placeholder=\"Введите адрес\"]').send_keys(adress)\n",
    "    sleep(2)\n",
    "    bot.driver.find_element(By.XPATH, '//ymaps[text()=\"Найти\"]').click()\n",
    "    sleep(2)\n",
    "    try:\n",
    "        bot.driver.find_element(By.XPATH, '//ymaps[contains(@class, \"__first\")]').click()\n",
    "    except:\n",
    "        pass\n",
    "    sleep(2)\n",
    "    bot.driver.find_element(By.XPATH, '//span[contains(text(), \"'+ adress +'\")]').click()\n",
    "    sleep(2)\n",
    "    bot.driver.find_element(By.XPATH, '//div[@class=\"balloon-content-block\"]/button').click()\n",
    "    sleep(2)\n",
    "    bot.driver.find_element(By.XPATH, '//button[@class=\"popup__btn-main\"]').click()\n",
    "    sleep(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "b485edae",
   "metadata": {},
   "outputs": [],
   "source": [
    "choose_post_place(post_places[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "44d9ebca",
   "metadata": {},
   "outputs": [],
   "source": [
    "import random"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "4e624ee0",
   "metadata": {},
   "outputs": [],
   "source": [
    "self = bot\n",
    "number = '9104499982'\n",
    "def get_qr_code(file_name):\n",
    "    try:\n",
    "        self.driver.find_element(By.XPATH, '//button[text()=\"                Оплатить заказ                \"]').click()\n",
    "        sleep(2)\n",
    "        self.driver.find_element(By.XPATH, '//button[contains(@class,\"popup__btn-main\")]').click()\n",
    "        sleep(2)\n",
    "        svg = self.driver.find_element(By.XPATH, '//div[@class=\"qr-code__value\"]')\n",
    "        self.save_qr_code(svg, file_name)\n",
    "        sleep(2)\n",
    "        self.driver.find_element(By.CLASS_NAME, 'popup__close').click()\n",
    "    except:\n",
    "#         Utils.login(number)\n",
    "        login(self.driver, number)\n",
    "        input(\"Нажми Enter, зайка\")\n",
    "        get_qr_code(file_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "022d4e25",
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "ename": "NoSuchElementException",
     "evalue": "Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\".j-main-login\"}\n  (Session info: chrome=100.0.4896.75)\nStacktrace:\nBacktrace:\n\tOrdinal0 [0x01039943+2595139]\n\tOrdinal0 [0x00FCC9F1+2148849]\n\tOrdinal0 [0x00EC4528+1066280]\n\tOrdinal0 [0x00EF0FD4+1249236]\n\tOrdinal0 [0x00EF11CB+1249739]\n\tOrdinal0 [0x00F1D812+1431570]\n\tOrdinal0 [0x00F0BA34+1358388]\n\tOrdinal0 [0x00F1BAF2+1424114]\n\tOrdinal0 [0x00F0B806+1357830]\n\tOrdinal0 [0x00EE6086+1204358]\n\tOrdinal0 [0x00EE6F96+1208214]\n\tGetHandleVerifier [0x011DB232+1658114]\n\tGetHandleVerifier [0x0129312C+2411516]\n\tGetHandleVerifier [0x010CF261+560433]\n\tGetHandleVerifier [0x010CE366+556598]\n\tOrdinal0 [0x00FD286B+2173035]\n\tOrdinal0 [0x00FD75F8+2192888]\n\tOrdinal0 [0x00FD76E5+2193125]\n\tOrdinal0 [0x00FE11FC+2232828]\n\tBaseThreadInitThunk [0x763BFA29+25]\n\tRtlGetAppContainerNamedObjectPath [0x77787A7E+286]\n\tRtlGetAppContainerNamedObjectPath [0x77787A4E+238]\n",
     "output_type": "error",
     "traceback": [
      "\u001B[1;31m---------------------------------------------------------------------------\u001B[0m",
      "\u001B[1;31mAttributeError\u001B[0m                            Traceback (most recent call last)",
      "\u001B[1;32mC:\\Users\\ASTROC~1\\AppData\\Local\\Temp/ipykernel_17272/2508058062.py\u001B[0m in \u001B[0;36mget_qr_code\u001B[1;34m(file_name)\u001B[0m\n\u001B[0;32m      9\u001B[0m         \u001B[0msvg\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mdriver\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mfind_element\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mBy\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mXPATH\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;34m'//div[@class=\"qr-code__value\"]'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m---> 10\u001B[1;33m         \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0msave_qr_code\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0msvg\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mfile_name\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m     11\u001B[0m         \u001B[0msleep\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;36m2\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;31mAttributeError\u001B[0m: 'Bot' object has no attribute 'save_qr_code'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001B[1;31mNoSuchElementException\u001B[0m                    Traceback (most recent call last)",
      "\u001B[1;32mC:\\Users\\ASTROC~1\\AppData\\Local\\Temp/ipykernel_17272/2136507517.py\u001B[0m in \u001B[0;36m<module>\u001B[1;34m\u001B[0m\n\u001B[1;32m----> 1\u001B[1;33m \u001B[0mget_qr_code\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;34m'file_name.svg'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m",
      "\u001B[1;32mC:\\Users\\ASTROC~1\\AppData\\Local\\Temp/ipykernel_17272/2508058062.py\u001B[0m in \u001B[0;36mget_qr_code\u001B[1;34m(file_name)\u001B[0m\n\u001B[0;32m     13\u001B[0m     \u001B[1;32mexcept\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     14\u001B[0m \u001B[1;31m#         Utils.login(number)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m---> 15\u001B[1;33m         \u001B[0mlogin\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mdriver\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mnumber\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m     16\u001B[0m         \u001B[0minput\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;34m\"Нажми Enter, зайка\"\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m     17\u001B[0m         \u001B[0mget_qr_code\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mfile_name\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\Users\\ASTROC~1\\AppData\\Local\\Temp/ipykernel_17272/2129243675.py\u001B[0m in \u001B[0;36mlogin\u001B[1;34m(driver, number)\u001B[0m\n\u001B[0;32m      1\u001B[0m \u001B[1;32mdef\u001B[0m \u001B[0mlogin\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mdriver\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mnumber\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m      2\u001B[0m         \u001B[1;31m# vars # 1\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m----> 3\u001B[1;33m         \u001B[0mto_login_btn\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mdriver\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mfind_element\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mBy\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mCLASS_NAME\u001B[0m\u001B[1;33m,\u001B[0m \u001B[1;34m'j-main-login'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m      4\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m      5\u001B[0m         \u001B[1;31m# resolve # 1\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001B[0m in \u001B[0;36mfind_element\u001B[1;34m(self, by, value)\u001B[0m\n\u001B[0;32m   1242\u001B[0m             \u001B[0mvalue\u001B[0m \u001B[1;33m=\u001B[0m \u001B[1;34m'[name=\"%s\"]'\u001B[0m \u001B[1;33m%\u001B[0m \u001B[0mvalue\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m   1243\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m-> 1244\u001B[1;33m         return self.execute(Command.FIND_ELEMENT, {\n\u001B[0m\u001B[0;32m   1245\u001B[0m             \u001B[1;34m'using'\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0mby\u001B[0m\u001B[1;33m,\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m   1246\u001B[0m             'value': value})['value']\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\u001B[0m in \u001B[0;36mexecute\u001B[1;34m(self, driver_command, params)\u001B[0m\n\u001B[0;32m    422\u001B[0m         \u001B[0mresponse\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mcommand_executor\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mexecute\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mdriver_command\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mparams\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    423\u001B[0m         \u001B[1;32mif\u001B[0m \u001B[0mresponse\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 424\u001B[1;33m             \u001B[0mself\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0merror_handler\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mcheck_response\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mresponse\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    425\u001B[0m             response['value'] = self._unwrap_value(\n\u001B[0;32m    426\u001B[0m                 response.get('value', None))\n",
      "\u001B[1;32mC:\\ProgramData\\Anaconda3\\lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py\u001B[0m in \u001B[0;36mcheck_response\u001B[1;34m(self, response)\u001B[0m\n\u001B[0;32m    245\u001B[0m                 \u001B[0malert_text\u001B[0m \u001B[1;33m=\u001B[0m \u001B[0mvalue\u001B[0m\u001B[1;33m[\u001B[0m\u001B[1;34m'alert'\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m.\u001B[0m\u001B[0mget\u001B[0m\u001B[1;33m(\u001B[0m\u001B[1;34m'text'\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    246\u001B[0m             \u001B[1;32mraise\u001B[0m \u001B[0mexception_class\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmessage\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mscreen\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mstacktrace\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0malert_text\u001B[0m\u001B[1;33m)\u001B[0m  \u001B[1;31m# type: ignore[call-arg]  # mypy is not smart enough here\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[1;32m--> 247\u001B[1;33m         \u001B[1;32mraise\u001B[0m \u001B[0mexception_class\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mmessage\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mscreen\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mstacktrace\u001B[0m\u001B[1;33m)\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0m\u001B[0;32m    248\u001B[0m \u001B[1;33m\u001B[0m\u001B[0m\n\u001B[0;32m    249\u001B[0m     \u001B[1;32mdef\u001B[0m \u001B[0m_value_or_default\u001B[0m\u001B[1;33m(\u001B[0m\u001B[0mself\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mobj\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0mMapping\u001B[0m\u001B[1;33m[\u001B[0m\u001B[0m_KT\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0m_VT\u001B[0m\u001B[1;33m]\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mkey\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0m_KT\u001B[0m\u001B[1;33m,\u001B[0m \u001B[0mdefault\u001B[0m\u001B[1;33m:\u001B[0m \u001B[0m_VT\u001B[0m\u001B[1;33m)\u001B[0m \u001B[1;33m->\u001B[0m \u001B[0m_VT\u001B[0m\u001B[1;33m:\u001B[0m\u001B[1;33m\u001B[0m\u001B[1;33m\u001B[0m\u001B[0m\n",
      "\u001B[1;31mNoSuchElementException\u001B[0m: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\".j-main-login\"}\n  (Session info: chrome=100.0.4896.75)\nStacktrace:\nBacktrace:\n\tOrdinal0 [0x01039943+2595139]\n\tOrdinal0 [0x00FCC9F1+2148849]\n\tOrdinal0 [0x00EC4528+1066280]\n\tOrdinal0 [0x00EF0FD4+1249236]\n\tOrdinal0 [0x00EF11CB+1249739]\n\tOrdinal0 [0x00F1D812+1431570]\n\tOrdinal0 [0x00F0BA34+1358388]\n\tOrdinal0 [0x00F1BAF2+1424114]\n\tOrdinal0 [0x00F0B806+1357830]\n\tOrdinal0 [0x00EE6086+1204358]\n\tOrdinal0 [0x00EE6F96+1208214]\n\tGetHandleVerifier [0x011DB232+1658114]\n\tGetHandleVerifier [0x0129312C+2411516]\n\tGetHandleVerifier [0x010CF261+560433]\n\tGetHandleVerifier [0x010CE366+556598]\n\tOrdinal0 [0x00FD286B+2173035]\n\tOrdinal0 [0x00FD75F8+2192888]\n\tOrdinal0 [0x00FD76E5+2193125]\n\tOrdinal0 [0x00FE11FC+2232828]\n\tBaseThreadInitThunk [0x763BFA29+25]\n\tRtlGetAppContainerNamedObjectPath [0x77787A7E+286]\n\tRtlGetAppContainerNamedObjectPath [0x77787A4E+238]\n"
     ]
    }
   ],
   "source": [
    "get_qr_code('file_name.svg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2f3c3198",
   "metadata": {},
   "outputs": [],
   "source": [
    "def login(driver, number):\n",
    "        # vars # 1\n",
    "        to_login_btn = driver.find_element(By.CLASS_NAME, 'j-main-login')\n",
    "\n",
    "        # resolve # 1\n",
    "        to_login_btn.click()\n",
    "        sleep(random.uniform(3, 5))\n",
    "        \n",
    "        # vars # 2\n",
    "        login_input = driver.find_element(By.CLASS_NAME, 'input-item')\n",
    "        login_btn = driver.find_element(By.CLASS_NAME, 'login__btn')\n",
    "\n",
    "        # resolve # 2\n",
    "        login_input.send_keys(number)\n",
    "        sleep(random.uniform(2, 4))\n",
    "        login_btn.click()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "[]+[{'a': '2'}]"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "def transform_price(price):\n",
    "    price = str(price)\n",
    "    if len(price)>=4:\n",
    "        price = price[:-2] + \"00\"\n",
    "    elif len(price)>1:\n",
    "        price = price[:-1] + \"0\"\n",
    "    return price"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "transform_price(12222)"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "! pip install ujson\n",
    "! pip install uvloop\n",
    "! pip install cchardet\n",
    "! pip install aiodns\n",
    "! pip install aiohttp[speedups]\n",
    "! pip install aiogram"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "import logging\n",
    "from aiogram import Bot, Dispatcher, executor, types\n"
   ],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false,
    "pycharm": {
     "name": "#%%\n"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "81a3f43d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'a': '2'}]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "b342b928",
   "metadata": {},
   "outputs": [],
   "source": [
    "def transform_price(price):\n",
    "    price = str(price)\n",
    "    if len(price)>=4:\n",
    "        price = price[:-2] + \"00\"\n",
    "    elif len(price)>1:\n",
    "        price = price[:-1] + \"0\"\n",
    "    return price"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "3f3b6c1c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'12200'"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "transform_price(12222)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ffda76bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: ujson in c:\\programdata\\anaconda3\\lib\\site-packages (4.0.2)\n",
      "Collecting uvloop"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c88c366010154bb38712f888667ddaa9\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c88c366010154bb38712f888667ddaa9\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-660ima5t'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c88c366010154bb38712f888667ddaa9\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c88c366010154bb38712f888667ddaa9\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ab/d9/22bbffa8f8d7e075ccdb29e8134107adfb4710feb10039f9d357db8b589c/uvloop-0.16.0.tar.gz#sha256=f74bc20c7b67d1c27c72601c78cf95be99d5c2cdd4514502b4f3eb0933ff1228 (from https://pypi.org/simple/uvloop/) (requires-python:>=3.7). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_a3da120275784d2e809d0eac9b37115d\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_a3da120275784d2e809d0eac9b37115d\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-7zfo9btu'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_a3da120275784d2e809d0eac9b37115d\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_a3da120275784d2e809d0eac9b37115d\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/41/49/d2cdac6450430f444d3df59ccd1abc8225ea51ffcecd9ac2e36b111074a4/uvloop-0.15.3.tar.gz#sha256=905f0adb0c09c9f44222ee02f6b96fd88b493478fffb7a345287f9444e926030 (from https://pypi.org/simple/uvloop/) (requires-python:>=3.7). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_8026e0be06be484fb739701d25b78618\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_8026e0be06be484fb739701d25b78618\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-8ojq6qzz'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_8026e0be06be484fb739701d25b78618\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_8026e0be06be484fb739701d25b78618\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/44/6e/0cb292e4e6ee1382e2ede458f90c94b4f990b261f738403ac45cb8183bc2/uvloop-0.15.2.tar.gz#sha256=2bb0624a8a70834e54dde8feed62ed63b50bad7a1265c40d6403a2ac447bce01 (from https://pypi.org/simple/uvloop/) (requires-python:>=3.7). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e0fe3465d9c0456baee16dcf8c4738ca\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e0fe3465d9c0456baee16dcf8c4738ca\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-335edvg4'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e0fe3465d9c0456baee16dcf8c4738ca\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e0fe3465d9c0456baee16dcf8c4738ca\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/84/2e/462e7a25b787d2b40cf6c9864a9e702f358349fc9cfb77e83c38acb73048/uvloop-0.14.0.tar.gz#sha256=123ac9c0c7dd71464f58f1b4ee0bbd81285d96cdda8bc3519281b8973e3a461e (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_970ea41fc1014f58b5add49e2bb55db9\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_970ea41fc1014f58b5add49e2bb55db9\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-9pk31vcq'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_970ea41fc1014f58b5add49e2bb55db9\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_970ea41fc1014f58b5add49e2bb55db9\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/e3/15/dc3276384f4363015d7c72282f37066bae26c77f99158f66c9058ac167cf/uvloop-0.13.0.tar.gz#sha256=942cd07035510b149d6160796f4e972137130ae953871b6a98c2cf5d5ab68c2e (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d5f12cf8d2604ebba92c8522b6f80197\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d5f12cf8d2604ebba92c8522b6f80197\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-dy95s157'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d5f12cf8d2604ebba92c8522b6f80197\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d5f12cf8d2604ebba92c8522b6f80197\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/75/99/7a470e4950d8ecb1b561e8483716c414cd51b8ee193de8eb23506275f5a7/uvloop-0.12.2.tar.gz#sha256=c48692bf4587ce281d641087658eca275a5ad3b63c78297bbded96570ae9ce8f (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_942c21d94df943f1a362d41a1dfbec3c\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_942c21d94df943f1a362d41a1dfbec3c\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-hck_zih4'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_942c21d94df943f1a362d41a1dfbec3c\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_942c21d94df943f1a362d41a1dfbec3c\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ec/74/96f1fdd6936d7a5ea2fa20bbd54b2c2242b1e92df425b37d00737982f2d7/uvloop-0.12.1.tar.gz#sha256=198fe0c196056930ec6c4a0a878e531a66d15467ca7c74a875aa90271f0c6e3f (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_2d8ea361c857448eb25188bf747f9e8e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_2d8ea361c857448eb25188bf747f9e8e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-017tntwp'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_2d8ea361c857448eb25188bf747f9e8e\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_2d8ea361c857448eb25188bf747f9e8e\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/f0/5a/cf8cb64bdc694462ef1ee9a925302dac40b29a52ab0eede4bab7fc0e0f2b/uvloop-0.12.0.tar.gz#sha256=afdf34bf507090e4c7f5108a17240982760356b8aae4edd37180ec4f94c36cbb (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_45ab2191b8dc459d99684272a84acc38\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_45ab2191b8dc459d99684272a84acc38\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-se_09r8j'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_45ab2191b8dc459d99684272a84acc38\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_45ab2191b8dc459d99684272a84acc38\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/46/60/8e46188acba22369848d5e9bcbee3ca04b68a12d0598bd625c1a3994f5df/uvloop-0.11.3.tar.gz#sha256=fd5042d0a2ea07b92d0e2190f7711feb91cde31cf2bf1829e2e8c4c0fdd1f1aa (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_bd35482ab8c042cba7a4516abd7c8d89\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_bd35482ab8c042cba7a4516abd7c8d89\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-w5vi59t_'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_bd35482ab8c042cba7a4516abd7c8d89\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_bd35482ab8c042cba7a4516abd7c8d89\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/5c/37/6daa39aac42b2deda6ee77f408bec0419b600e27b89b374b0d440af32b10/uvloop-0.11.2.tar.gz#sha256=a97bd62ebbdf7e6e84bf44afe439d9b24ce4d8661a29a639626a8c03748f6f98 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c9850125b4f44dcc897e148f632ec64e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c9850125b4f44dcc897e148f632ec64e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-lk1vitaf'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c9850125b4f44dcc897e148f632ec64e\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c9850125b4f44dcc897e148f632ec64e\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/c7/87/39c3a2a91c2079fa43f3a49fe3423eacc4c4ab2247a8419a80f2a2626cde/uvloop-0.11.1.tar.gz#sha256=54be0ce6213e5488fd2332ae52f4ff20800aaf9bfe3d08f3422bc71a035903c8 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_36d1bb8aff7e4d2cb4cd14ee9b9e08f4\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_36d1bb8aff7e4d2cb4cd14ee9b9e08f4\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-t88abyp4'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_36d1bb8aff7e4d2cb4cd14ee9b9e08f4\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_36d1bb8aff7e4d2cb4cd14ee9b9e08f4\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/cd/e5/1492d3a6ee52f261df764eece684513bb8abbe988d4bd3d94093f39e2bea/uvloop-0.11.0.tar.gz#sha256=34138af5f6ced8f2f2ba39b487119988f5055245223a330de71957c8a6713b84 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_0062915acfeb47968afd93bc935eb446\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_0062915acfeb47968afd93bc935eb446\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-9dr7iz9s'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_0062915acfeb47968afd93bc935eb446\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_0062915acfeb47968afd93bc935eb446\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/69/c6/5a6967f9718b60732dd41b3bea02b2b7d3ebfa40281eee0d8d5437b530c4/uvloop-0.10.3.tar.gz#sha256=0cd8b104f962f9ba01f3a64bfb1c941fdb99819fd60a8b8949244e651eec007a (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_12dfaaaef04f490abdbef9e329b135a1\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_12dfaaaef04f490abdbef9e329b135a1\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-7nw_9sgq'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_12dfaaaef04f490abdbef9e329b135a1\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_12dfaaaef04f490abdbef9e329b135a1\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/34/ee/624682b83e8b5910ac0b1135f18944ffd8cb82354f3c481cc523f9d6b627/uvloop-0.10.2.tar.gz#sha256=b7dc00fe9d0a04365d4c98f40316c8913463dc5b57ab64f23282d2e1d5c2defe (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_71e9f7822c284c7ca782252997764d0d\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_71e9f7822c284c7ca782252997764d0d\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-iu7500i5'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_71e9f7822c284c7ca782252997764d0d\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_71e9f7822c284c7ca782252997764d0d\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/d4/50/385f5dfd008508750a255ae742aaea6cc7f58877d9444b2d759543b0ee72/uvloop-0.10.1.tar.gz#sha256=90f9d3ba56ddd554954feff381306ec7e8c31305fc8bf686277b7d8c46e09d68 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_7cc2c184e7f9446290bd23bfa9369e0c\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_7cc2c184e7f9446290bd23bfa9369e0c\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-i7urjpd7'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_7cc2c184e7f9446290bd23bfa9369e0c\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_7cc2c184e7f9446290bd23bfa9369e0c\\setup.py\", line 15, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/eb/1e/0012c2e56a49df0c7c604e1eb5dce05b4e86de705bd5c49745c1d6a50dc5/uvloop-0.10.0.tar.gz#sha256=f9cd68448e45f5a609d2d679deb3d434be9ab68a4cc32e24c5eb9309e3e3f9db (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ebb0b202da1840caabceba040f1ab8b3\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ebb0b202da1840caabceba040f1ab8b3\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-9fofw45y'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ebb0b202da1840caabceba040f1ab8b3\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ebb0b202da1840caabceba040f1ab8b3\\setup.py\", line 10, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/f3/e0/05c5c2a879b3ee8427a43ca2abd34cffc172e2cb5c45217d67e06b36a48e/uvloop-0.9.1.tar.gz#sha256=7ee14835a75c72227d3f8a3f370519a3106a6f02e5453f275f16437ebdb92953 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e30de41f2f094a708b345b6d758eec5c\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e30de41f2f094a708b345b6d758eec5c\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-pghp9tlf'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e30de41f2f094a708b345b6d758eec5c\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e30de41f2f094a708b345b6d758eec5c\\setup.py\", line 10, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/2f/d2/ace6713b6ec2aed8a6d6732dc29613b26ae8c0676b57713f92fda8045579/uvloop-0.9.0.tar.gz#sha256=a15b96a2139ca6ceb779397c08a1089955e6f3308b030883aa4dcfda875f0a10 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ae48bf72d3d7460c84057e4e29c4d4a0\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ae48bf72d3d7460c84057e4e29c4d4a0\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-k3wqybpf'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ae48bf72d3d7460c84057e4e29c4d4a0\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ae48bf72d3d7460c84057e4e29c4d4a0\\setup.py\", line 10, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/8a/90/2ee7d33ed33cdd493f1f2e78ca2c3cb7823e2dceb875ac52e7096d4f6004/uvloop-0.8.1.tar.gz#sha256=923b6e4982aea14a23565e6e62335a691b7a939bcc6802fbafa2a146ee3e8ac3 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9f5d3a099a50447d8c50a42661b2e167\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9f5d3a099a50447d8c50a42661b2e167\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-m2tszmcc'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9f5d3a099a50447d8c50a42661b2e167\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9f5d3a099a50447d8c50a42661b2e167\\setup.py\", line 11, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/30/d4/3a85abecaea42c1ca4d2a23c62fe2b94017370c99abec38f6d70b38d0e9f/uvloop-0.8.0.tar.gz#sha256=2c22f1f0b6e00e99f5ee635993e5f26418e740bd26f1214cf78c70f237eec06b (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9899f837dae74a34ba7d924e09250652\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9899f837dae74a34ba7d924e09250652\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-frxmj2rj'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9899f837dae74a34ba7d924e09250652\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9899f837dae74a34ba7d924e09250652\\setup.py\", line 11, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/6d/2f/d9d64e467f4b0bee5ac2ba0ae14a728956e4069711af1777db88f3d909c1/uvloop-0.7.2.tar.gz#sha256=9bbf231e93b08c8411a0aa0b8fb510c565096b548afa4d1efff7e150990d3845 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_db0b2f3d1a8a4cddae4dce3f9be0f188\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_db0b2f3d1a8a4cddae4dce3f9be0f188\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-qfv4rynb'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_db0b2f3d1a8a4cddae4dce3f9be0f188\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_db0b2f3d1a8a4cddae4dce3f9be0f188\\setup.py\", line 11, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/03/14/addd76b562029236adfbf150f38af1296549892f89acb4168f40c361c1e1/uvloop-0.7.1.tar.gz#sha256=e606cc9de2036430e5c502a58e265a7788134096803349ff0c537ec65044ef12 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_cb96fd4b857d44d788add6f7fb950763\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_cb96fd4b857d44d788add6f7fb950763\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-4ska2tko'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_cb96fd4b857d44d788add6f7fb950763\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_cb96fd4b857d44d788add6f7fb950763\\setup.py\", line 11, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ce/67/706cd6b1d0a01abb491dff22ed05ad6f3bbfd792edf7cc9731f9cc30ae2a/uvloop-0.7.0.tar.gz#sha256=e94944ce0f35858dcad084aa396d25a54c21383a207171d416908f609aa382df (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c142dd29949044c8b46d93129c9e9b67\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c142dd29949044c8b46d93129c9e9b67\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-xufwk17f'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c142dd29949044c8b46d93129c9e9b67\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c142dd29949044c8b46d93129c9e9b67\\setup.py\", line 10, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/51/b2/9319744df7615915c6dd176de9387c8b272a02049cebd914069a4954e835/uvloop-0.6.8.tar.gz#sha256=36dd35b1931576ff9c95a7aa1d62ea8921139982bf36d607f35c79cff6915343 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_636b01f06cc3400fabc5a5889886fd4e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_636b01f06cc3400fabc5a5889886fd4e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-9tg2fow5'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_636b01f06cc3400fabc5a5889886fd4e\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_636b01f06cc3400fabc5a5889886fd4e\\setup.py\", line 10, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/01/ac/164734940b4f170cd40d567aa92176893e7ff0be091ae1994feb6e1b939c/uvloop-0.6.7.tar.gz#sha256=843bc46f13bc75367e80bbc2c9a8bb3f2fa3ac6ee5bc4bf72fdf3c23dda00a38 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9d423e4dbbdc4d928693d10b71d008f5\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9d423e4dbbdc4d928693d10b71d008f5\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-d853yxju'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9d423e4dbbdc4d928693d10b71d008f5\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9d423e4dbbdc4d928693d10b71d008f5\\setup.py\", line 10, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/8d/26/5c9c7bb9193f5665ad057b31e927f749f968bd89ec089e76c6aa14e46d9c/uvloop-0.6.6.tar.gz#sha256=fa0268b8dfd8af4d1002d6d15f9d8ffe98f3aa424b2bb6cb30c4858385bb726d (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_fd780ff655414e4b88228a0f14b68af6\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_fd780ff655414e4b88228a0f14b68af6\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-9omxi4kp'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_fd780ff655414e4b88228a0f14b68af6\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_fd780ff655414e4b88228a0f14b68af6\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/78/f7/9a1ee6b9576e29296355147c60e641263b4c12c2e6146cc032a5cc1b0861/uvloop-0.6.5.tar.gz#sha256=bfe0ffd5ec7d3ef8714fe4db089ae8c757e7260fa9ef67ee577a03d6d0f9d57a (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c4195ab960fb46f88d6850891bc7b736\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c4195ab960fb46f88d6850891bc7b736\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-t0wz_f8q'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c4195ab960fb46f88d6850891bc7b736\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c4195ab960fb46f88d6850891bc7b736\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/8f/a9/fbb7da67e9a88b95e4d2666c4156fd485240e8dd81b21291a1066786b89a/uvloop-0.6.0.tar.gz#sha256=506ec1d1139fd10793d0d16f6dc362df5b5e1615ec1b073e617fe7b628ea250e (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_f2ad1e9ecad54ca6bab22c04c65c34c9\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_f2ad1e9ecad54ca6bab22c04c65c34c9\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-bw4qxz6p'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_f2ad1e9ecad54ca6bab22c04c65c34c9\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_f2ad1e9ecad54ca6bab22c04c65c34c9\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/09/38/2aaacc50a7f44385a9076ef90134b21a442518801e26178aec7dd9210a14/uvloop-0.5.5.tar.gz#sha256=43eb59f489acb2d2b7715ff228b104922b06947be97ff0b38e7a662dc6d4199d (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_cf14d689580144d688fe823e6d93aec5\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_cf14d689580144d688fe823e6d93aec5\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-ph4q5trh'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_cf14d689580144d688fe823e6d93aec5\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_cf14d689580144d688fe823e6d93aec5\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/54/60/a7511c700352e7de93d824bd7ef092a1d3644431a46d9b0c1366138a7c71/uvloop-0.5.4.tar.gz#sha256=06db01a80db610787fccd241aeaa45731b587ae0aeb894d0098a8a710463ce3f (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e6b9395c19954b09b5629a6a39abc788\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e6b9395c19954b09b5629a6a39abc788\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-fkumva_c'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e6b9395c19954b09b5629a6a39abc788\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e6b9395c19954b09b5629a6a39abc788\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/ed/02/aa66d1507c372ae35533e72bdb5e1f6cdf77ef2cd7aa0ffae7b0e0c3d144/uvloop-0.5.3.tar.gz#sha256=ee402d2df4bf092f6f036637db41bb2896f7c6832adf91dbb6dfec0e6dfa6aee (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_94a4710ce1b840d2a610fa48d73b8ca4\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_94a4710ce1b840d2a610fa48d73b8ca4\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-gsrz6weg'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_94a4710ce1b840d2a610fa48d73b8ca4\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_94a4710ce1b840d2a610fa48d73b8ca4\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/4b/29/67ce259e01ad98c97a891bece37e324d1f5935aec0d32bbff3f432f04a56/uvloop-0.5.2.tar.gz#sha256=f328c56e92b0a111b48faf1cc4fde616c4c0eb733d20f8c298ac69dce52b5ed2 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ae26396ddbf4472f849d41dbf5455620\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ae26396ddbf4472f849d41dbf5455620\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-16y3_4il'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ae26396ddbf4472f849d41dbf5455620\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ae26396ddbf4472f849d41dbf5455620\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/7d/6b/096dc6a28655d8038ac6f0f212dffd2119ca7465226f1fd6ba2c902d35a2/uvloop-0.5.1.tar.gz#sha256=dca6364c5e74ab71cdb362d417f263cfe21c87f722ce21dc7f80b3541d4fd10a (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_376dac41db5949dc8dc55ab7fa9d8456\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_376dac41db5949dc8dc55ab7fa9d8456\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-bdovf8zj'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_376dac41db5949dc8dc55ab7fa9d8456\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_376dac41db5949dc8dc55ab7fa9d8456\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/86/fc/cdf05f20f9257300f1eb91c3ea196e392e7876552c82b83d65a382e2d8ef/uvloop-0.5.0.tar.gz#sha256=acb43345d0ead57025f66393b73e4cb4a536a43a7fc6bd8b0fc8f2bad8eaf33e (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_89c94176f412439098702dd5054f10ae\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_89c94176f412439098702dd5054f10ae\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-2qaphfri'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_89c94176f412439098702dd5054f10ae\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_89c94176f412439098702dd5054f10ae\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/fa/f6/06eff3c4e67a80007051c4fc23d54edbe202fcd0fcea3a3c8237908d4dbb/uvloop-0.4.34.tar.gz#sha256=93c4b7b2c209f1bb6a8cea9553171153ceca72a94c094bf2e4460e5671ae016f (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_2b74f04a99d845a89479b327b7a05a69\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_2b74f04a99d845a89479b327b7a05a69\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-hip7utm0'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_2b74f04a99d845a89479b327b7a05a69\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_2b74f04a99d845a89479b327b7a05a69\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/db/e0/497236facd90d0e19c051a2fc5f2f6da72d875805e04e3a717a726882cd7/uvloop-0.4.33.tar.gz#sha256=1cf8cf3a372c83e6397f88884f4bcd6afa3dbe7f8fa3df1dfe98407d3d3a6c4b (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d0a656380c794b8bb3e39036aa877c7d\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d0a656380c794b8bb3e39036aa877c7d\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-7k5b_p9k'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d0a656380c794b8bb3e39036aa877c7d\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d0a656380c794b8bb3e39036aa877c7d\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/7a/41/823341aca45dc3524beb1d1821d72a696a2de3d4f0f08d42ee5933a91db1/uvloop-0.4.32.tar.gz#sha256=d7788f1e71e29955cef80ef0635dc484ca5a736f8c01db751740ba10df245ede (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_1fb53551755d422c95f20e04dada21bd\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_1fb53551755d422c95f20e04dada21bd\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-kpq7wc3i'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_1fb53551755d422c95f20e04dada21bd\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_1fb53551755d422c95f20e04dada21bd\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/cb/d9/9871e573c30f414e89a4af83c7866bba6a9d4f9df560249da19db9abde17/uvloop-0.4.31.tar.gz#sha256=25a11dcc61062d9e39faa5a7a3713616fae56e7841ad14b5a31257a272d2822b (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_db671197836142578cc4c6423d7f7c24\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_db671197836142578cc4c6423d7f7c24\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-m7s4gdph'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_db671197836142578cc4c6423d7f7c24\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_db671197836142578cc4c6423d7f7c24\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/81/2d/27e695ec3fb1eb4e1c8676ee8caae38a26723a4e723dae00ac215f41dfb1/uvloop-0.4.30.tar.gz#sha256=62d95872acd9ea022d73cf2c47af1eab51b47f4cf7917ece419769d68a0d6320 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ef200daad39e45e2a67eaa85678f313f\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_ef200daad39e45e2a67eaa85678f313f\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-vt46i365'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ef200daad39e45e2a67eaa85678f313f\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_ef200daad39e45e2a67eaa85678f313f\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/95/8b/df43909e26c8b6e1edf46bd3383a50120c85fb81af470c7c19b8c4672992/uvloop-0.4.29.tar.gz#sha256=08576d9e8da48eac7af8f42087357211fcdeb3bf547ccd4b20be6b2b7d70c0da (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d1e6b36f816c4063803bb6be1815cd93\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d1e6b36f816c4063803bb6be1815cd93\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-zqw6l04j'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d1e6b36f816c4063803bb6be1815cd93\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d1e6b36f816c4063803bb6be1815cd93\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/a5/a8/ae70fb1e73f239c7faebff0a7bcfa6570214b178197cfce837d24a753dd8/uvloop-0.4.28.tar.gz#sha256=94b6a18e5e3272213a74c0876d12532c14a521602186c010c3a948be07a2726e (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_da79ff5a6bb448878169bc78020957b9\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_da79ff5a6bb448878169bc78020957b9\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-m0omksx7'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_da79ff5a6bb448878169bc78020957b9\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_da79ff5a6bb448878169bc78020957b9\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/54/47/b49a47ee825a59e6ee2e43a177a1787a06c7ca3bde7fb1a1be0dd035a370/uvloop-0.4.27.tar.gz#sha256=ee0925027c2f8b53aed09c427745b9f96d8ead6eb2d333298ce3f41f0f084b14 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e6ea9ed71a9847afa2c4e624528322cb\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_e6ea9ed71a9847afa2c4e624528322cb\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-v40ckdgi'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e6ea9ed71a9847afa2c4e624528322cb\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_e6ea9ed71a9847afa2c4e624528322cb\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b3/c0/aa02843f0b24d10604a402957b871292ea0717716961f9d2eacdc8ca1334/uvloop-0.4.26.tar.gz#sha256=4f6ca1555b9492f4b44a0de6ac3a77a60e9d63e0b97f410b11dccbb282aed79a (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_101f881b1e474917b661957fd3db1149\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_101f881b1e474917b661957fd3db1149\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-um0zlxsg'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_101f881b1e474917b661957fd3db1149\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_101f881b1e474917b661957fd3db1149\\setup.py\", line 9, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/e6/45/413bdf83bf61975f2602e1f1844c1c7f9f4196f78ae4c075839ac06367ea/uvloop-0.4.25.tar.gz#sha256=22fb1ba75f6d35a36fbfe2b67537948cfe476eda18096f3acdd1f903f99fb9e9 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_f059b297da7a46909648c6649102b571\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_f059b297da7a46909648c6649102b571\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-157p75qd'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_f059b297da7a46909648c6649102b571\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_f059b297da7a46909648c6649102b571\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/82/05/a162fbbd4213fb844f1ab8f2dc18f285dd507fcb440742466bc8dcac404f/uvloop-0.4.24.tar.gz#sha256=3ba853b920d9c7ef78e2898adb4c0cc252e5c41ab793ccb230cc99c3995b45d4 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_bba4790e6447494e844255904fb24358\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_bba4790e6447494e844255904fb24358\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-otlfmdhx'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_bba4790e6447494e844255904fb24358\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_bba4790e6447494e844255904fb24358\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/94/6f/6308c4ed20145892341d545212957567e10e459af69c87a934c6c088411f/uvloop-0.4.23.tar.gz#sha256=47c100da05a2906878827656aea4f584375d6939060de32f86b2f172fe6c177c (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_405fe41cc3e841baa0ef9aff37202664\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_405fe41cc3e841baa0ef9aff37202664\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-q15yourk'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_405fe41cc3e841baa0ef9aff37202664\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_405fe41cc3e841baa0ef9aff37202664\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/27/b8/c3233433b5b90c712253a280d4e5cc8ccd9fb115e1c30a2eb444f9d732bd/uvloop-0.4.22.tar.gz#sha256=a1ce233f28c17452680f6281e5efdd7fbd6d866a448a82923e7e1065ee290935 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d65497ccbb004b5fbb63bd4edd18e806\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_d65497ccbb004b5fbb63bd4edd18e806\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-tjkm_n2s'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d65497ccbb004b5fbb63bd4edd18e806\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_d65497ccbb004b5fbb63bd4edd18e806\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/99/ca/906342cc64cf7bc898878d504394f885627cd5baa6f160cd6cae335cd234/uvloop-0.4.21.tar.gz#sha256=aca14bf99c1dd3298803f4c7739e9d8ee1828795266ed06a2c31e48ce9e7e517 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_7aeba4e082e247e881743f8d24a48504\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_7aeba4e082e247e881743f8d24a48504\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-9c9eets2'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_7aeba4e082e247e881743f8d24a48504\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_7aeba4e082e247e881743f8d24a48504\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/c5/85/e5d62e52094053cf2ad12a4811ab6f68ef7ae5312bf431ef750cc5dc2b7a/uvloop-0.4.20.tar.gz#sha256=b39afbc242eafff7471337468b28c4164f55d130f8194e0fb4a8b2eb71cb1298 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_0b60cfe393c742429b68fbf07acfc7d4\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_0b60cfe393c742429b68fbf07acfc7d4\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-w791bdzn'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_0b60cfe393c742429b68fbf07acfc7d4\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_0b60cfe393c742429b68fbf07acfc7d4\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/34/17/08d8849f8dc303165d10db4889bb71c35e54ad04df3ddc99118a8e8a1759/uvloop-0.4.19.tar.gz#sha256=4853e43e7c4c06e5d32f8aec5e823e0949aa1f3f080bd998c73bcf9294ac372c (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_44b0203b9b1e48b39cfdf5f18f160100\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_44b0203b9b1e48b39cfdf5f18f160100\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-xzf88pyo'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_44b0203b9b1e48b39cfdf5f18f160100\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_44b0203b9b1e48b39cfdf5f18f160100\\setup.py\", line 8, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/33/1e/0dd8cf3d64b20f3ae821c661199fe53bb4ffa0fd24e8344c10adc0b07700/uvloop-0.4.18.tar.gz#sha256=181141e622da6138f281e9269af2704db60731e87291b8ffd2cda0dbb289c404 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_097b74622b134681ba251d4e8a3c4b3e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_097b74622b134681ba251d4e8a3c4b3e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-ou2q_pqi'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_097b74622b134681ba251d4e8a3c4b3e\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_097b74622b134681ba251d4e8a3c4b3e\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/6f/da/df5d69a91a15b9fc3825f229be4a146bb959cc168e55332ea154bebf7064/uvloop-0.4.17.tar.gz#sha256=66f082b8f04ab4e3f4484da973ca07978c9c8584ce3f5b85350953e34f6a8970 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_f758e0264e6741218be8c76b73010a98\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_f758e0264e6741218be8c76b73010a98\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-8dc1iso8'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_f758e0264e6741218be8c76b73010a98\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_f758e0264e6741218be8c76b73010a98\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/8e/d7/7ebafcd8e999f994a9bbb51d4ac0a9e0e215dda391c92888ec70f7161bbd/uvloop-0.4.16.tar.gz#sha256=0a2e9035b863276b3df7f270fb52a27532e63c22b0b726289b466f73c8b7af9d (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_90b07666459345928989690ab728fe19\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_90b07666459345928989690ab728fe19\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-49koue_l'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_90b07666459345928989690ab728fe19\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_90b07666459345928989690ab728fe19\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/4e/50/18e5b4b807771918ef639e237dc7a52b0f8baaf3a6d0fe97c1d982261860/uvloop-0.4.15.tar.gz#sha256=39027ef43a0102c0e3437edcbbd9aee0515c0836089de1760467fc21733e8bf9 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c3005e09d4534213b2c0a367f80aecee\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c3005e09d4534213b2c0a367f80aecee\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-7v423r4v'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c3005e09d4534213b2c0a367f80aecee\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c3005e09d4534213b2c0a367f80aecee\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/c1/f9/a95e183fece7200a071d3029587ba5ce4cb1877e266e6bc9b9bc086771dd/uvloop-0.4.14.tar.gz#sha256=484f975a987b7c8180b02754535573d15389318408d4bd121551e4ec446a250c (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_a8ad4f7068d94a3397593363af5a082c\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_a8ad4f7068d94a3397593363af5a082c\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-qsk0dx1_'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_a8ad4f7068d94a3397593363af5a082c\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_a8ad4f7068d94a3397593363af5a082c\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/7f/50/68ab5114d7d82f65ad4db888405a7838ebb45fcdf0db2ab6aed419013fae/uvloop-0.4.13.tar.gz#sha256=46ac1d2d66c217b770bf7086fa6fce9b16f506655e788df9634fa24b028cf6a3 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_cf5cd4c6880c406cb358c42fccff11b2\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_cf5cd4c6880c406cb358c42fccff11b2\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-16frtl4l'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_cf5cd4c6880c406cb358c42fccff11b2\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_cf5cd4c6880c406cb358c42fccff11b2\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/75/49/34adb38fba0fd6609e12af6b395357e069aa2f6d90f7c2c91dd9f0dd53d3/uvloop-0.4.12.tar.gz#sha256=bff6086ca164908d5b1c0db9bf33f8c87b0334a04aeb4773c542224fdfa4b933 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c45acf188a23459fb09fe8495859419a\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_c45acf188a23459fb09fe8495859419a\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-pbyyj9wc'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c45acf188a23459fb09fe8495859419a\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_c45acf188a23459fb09fe8495859419a\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/39/2a/561a11103dde5ea77609f9f5f251fc290afee22b448ba68ddd2e21f194db/uvloop-0.4.11.tar.gz#sha256=7e0e764dc3c92aece6ec7638c6ececfad29c7e3244a7ddc7746d521a34f9aa83 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_958e2a0d02fe4c0eaa7563a7ff757d3e\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_958e2a0d02fe4c0eaa7563a7ff757d3e\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-tc5ayo9s'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_958e2a0d02fe4c0eaa7563a7ff757d3e\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_958e2a0d02fe4c0eaa7563a7ff757d3e\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/93/06/c63b046f822e5c134bd740b11eff22630db3a21452b854adc2294daa1dbd/uvloop-0.4.10.tar.gz#sha256=2e278686bc02cc89658730c497a21ed9057464611e46b9b1ccadc074b49b16ac (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_be9e06f7ea564ddb90a3bd7e0e739af2\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_be9e06f7ea564ddb90a3bd7e0e739af2\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-6zj4csrr'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_be9e06f7ea564ddb90a3bd7e0e739af2\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_be9e06f7ea564ddb90a3bd7e0e739af2\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/31/1c/68d66aeea88e81793ed823c88af2125fbffbf06ac3c10f619064fe20524e/uvloop-0.4.9.tar.gz#sha256=e399fa2a03be2544313cbaae3a5b1ab0cf813330da26334d80aa4eb927e086e0 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_6226636ae01044b78016a04c09fb1d68\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_6226636ae01044b78016a04c09fb1d68\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-972wedvo'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_6226636ae01044b78016a04c09fb1d68\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_6226636ae01044b78016a04c09fb1d68\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/b0/37/44e73e20bac980aa736db2dc756ed2af2991da5a56650165a834f6cae639/uvloop-0.4.8.tar.gz#sha256=c1eb3a1108b5706f4d8c25bca24e2f93d57f3411d938983e3ec06ca5efb5403c (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9576e57f1bb547569f0e03b79fb1d544\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9576e57f1bb547569f0e03b79fb1d544\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-6vfda5h3'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9576e57f1bb547569f0e03b79fb1d544\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9576e57f1bb547569f0e03b79fb1d544\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/1d/67/2538cbb4abb717f35d84e7f42581091c6034e313602ad0b0ef0272424603/uvloop-0.4.7.tar.gz#sha256=6cdbdfe85778654ae74683ee936e7fac5f091cf1f1e1be561b407b94f3ecb6b5 (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_37ee9131225d46d5b51611f7bc1dfd55\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_37ee9131225d46d5b51611f7bc1dfd55\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-0go5yh4c'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_37ee9131225d46d5b51611f7bc1dfd55\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_37ee9131225d46d5b51611f7bc1dfd55\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/83/64/da8b55c689ab12cc29a7750b40c6d373e41228fdf53301885a6b9ffdc105/uvloop-0.4.6.tar.gz#sha256=8331ba2d5d4f1bebeb84c31efe9e6e0a23b18c273bee0a63096362da1de9519f (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_83d30aa75afd454381276e110dfed2a7\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_83d30aa75afd454381276e110dfed2a7\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' egg_info --egg-base 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-pip-egg-info-5dfegoiw'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_83d30aa75afd454381276e110dfed2a7\\\n",
      "    Complete output (5 lines):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 1, in <module>\n",
      "      File \"C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_83d30aa75afd454381276e110dfed2a7\\setup.py\", line 7, in <module>\n",
      "        raise RuntimeError('uvloop does not support Windows at the moment')\n",
      "    RuntimeError: uvloop does not support Windows at the moment\n",
      "    ----------------------------------------\n",
      "WARNING: Discarding https://files.pythonhosted.org/packages/8b/a5/781a73367f2e00a01bfd9d501bfb255302dcd09927c74602704e22ac1a15/uvloop-0.4.5.tar.gz#sha256=0c92ee0de7b64bef0c00235ffc54c62d209a4ad252ed144330193e6413898a4a (from https://pypi.org/simple/uvloop/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.\n",
      "  ERROR: Command errored out with exit status 1:\n",
      "   command: 'C:\\ProgramData\\Anaconda3\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9e9339644a334937bca99aa1160d2790\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9e9339644a334937bca99aa1160d2790\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' bdist_wheel -d 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-wheel-m8paujxt'\n",
      "       cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9e9339644a334937bca99aa1160d2790\\\n",
      "  Complete output (61 lines):\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_py\n",
      "  creating build\n",
      "  creating build\\lib.win-amd64-3.9\n",
      "  creating build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\_testbase.py -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\__init__.py -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  running egg_info\n",
      "  writing uvloop.egg-info\\PKG-INFO\n",
      "  writing dependency_links to uvloop.egg-info\\dependency_links.txt\n",
      "  writing top-level names to uvloop.egg-info\\top_level.txt\n",
      "  reading manifest file 'uvloop.egg-info\\SOURCES.txt'\n",
      "  reading manifest template 'MANIFEST.in'\n",
      "  adding license file 'LICENSE'\n",
      "  writing manifest file 'uvloop.egg-info\\SOURCES.txt'\n",
      "  copying uvloop\\cbhandles.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\cbhandles.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\dns.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\errors.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\loop.c -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\loop.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\loop.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\os_signal.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\os_signal.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\request.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\request.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\server.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  copying uvloop\\server.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "  creating build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\async_.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\async_.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\handle.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\handle.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\idle.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\idle.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\pipe.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\pipe.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\poll.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\poll.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\process.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\process.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\signal.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\signal.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\stream.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\stream.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\tcp.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\tcp.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\timer.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\timer.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\transport.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  copying uvloop\\handles\\transport.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "  creating build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  copying uvloop\\includes\\__init__.py -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  copying uvloop\\includes\\consts.pxi -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  copying uvloop\\includes\\python.pxd -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  copying uvloop\\includes\\stdlib.pxi -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  copying uvloop\\includes\\system.pxd -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  copying uvloop\\includes\\uv.pxd -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "  running build_ext\n",
      "  error: [WinError 2] Не удается найти указанный файл\n",
      "  ----------------------------------------\n",
      "  ERROR: Failed building wheel for uvloop\n",
      "    ERROR: Command errored out with exit status 1:\n",
      "     command: 'C:\\ProgramData\\Anaconda3\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9e9339644a334937bca99aa1160d2790\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9e9339644a334937bca99aa1160d2790\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-record-2oq72hhd\\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\\ProgramData\\Anaconda3\\Include\\uvloop'\n",
      "         cwd: C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-install-0iy9qm68\\uvloop_9e9339644a334937bca99aa1160d2790\\\n",
      "    Complete output (61 lines):\n",
      "    running install\n",
      "    running build\n",
      "    running build_py\n",
      "    creating build\n",
      "    creating build\\lib.win-amd64-3.9\n",
      "    creating build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\_testbase.py -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\__init__.py -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    running egg_info\n",
      "    writing uvloop.egg-info\\PKG-INFO\n",
      "    writing dependency_links to uvloop.egg-info\\dependency_links.txt\n",
      "    writing top-level names to uvloop.egg-info\\top_level.txt\n",
      "    reading manifest file 'uvloop.egg-info\\SOURCES.txt'\n",
      "    reading manifest template 'MANIFEST.in'\n",
      "    adding license file 'LICENSE'\n",
      "    writing manifest file 'uvloop.egg-info\\SOURCES.txt'\n",
      "    copying uvloop\\cbhandles.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\cbhandles.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\dns.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\errors.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\loop.c -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\loop.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\loop.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\os_signal.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\os_signal.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\request.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\request.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\server.pxd -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    copying uvloop\\server.pyx -> build\\lib.win-amd64-3.9\\uvloop\n",
      "    creating build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\async_.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\async_.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\handle.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\handle.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\idle.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\idle.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\pipe.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\pipe.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\poll.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\poll.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\process.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\process.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\signal.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\signal.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\stream.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\stream.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\tcp.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\tcp.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\timer.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\timer.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\transport.pxd -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    copying uvloop\\handles\\transport.pyx -> build\\lib.win-amd64-3.9\\uvloop\\handles\n",
      "    creating build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    copying uvloop\\includes\\__init__.py -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    copying uvloop\\includes\\consts.pxi -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    copying uvloop\\includes\\python.pxd -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    copying uvloop\\includes\\stdlib.pxi -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    copying uvloop\\includes\\system.pxd -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    copying uvloop\\includes\\uv.pxd -> build\\lib.win-amd64-3.9\\uvloop\\includes\n",
      "    running build_ext\n",
      "    error: [WinError 2] Не удается найти указанный файл\n",
      "    ----------------------------------------\n",
      "ERROR: Command errored out with exit status 1: 'C:\\ProgramData\\Anaconda3\\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9e9339644a334937bca99aa1160d2790\\\\setup.py'\"'\"'; __file__='\"'\"'C:\\\\Users\\\\Astrocatling\\\\AppData\\\\Local\\\\Temp\\\\pip-install-0iy9qm68\\\\uvloop_9e9339644a334937bca99aa1160d2790\\\\setup.py'\"'\"';f = getattr(tokenize, '\"'\"'open'\"'\"', open)(__file__) if os.path.exists(__file__) else io.StringIO('\"'\"'from setuptools import setup; setup()'\"'\"');code = f.read().replace('\"'\"'\\r\\n'\"'\"', '\"'\"'\\n'\"'\"');f.close();exec(compile(code, __file__, '\"'\"'exec'\"'\"'))' install --record 'C:\\Users\\Astrocatling\\AppData\\Local\\Temp\\pip-record-2oq72hhd\\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\\ProgramData\\Anaconda3\\Include\\uvloop' Check the logs for full command output.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  Downloading uvloop-0.16.0.tar.gz (2.1 MB)\n",
      "  Downloading uvloop-0.15.3.tar.gz (2.1 MB)\n",
      "  Downloading uvloop-0.15.2.tar.gz (2.1 MB)\n",
      "  Downloading uvloop-0.14.0.tar.gz (2.0 MB)\n",
      "  Downloading uvloop-0.13.0.tar.gz (2.0 MB)\n",
      "  Downloading uvloop-0.12.2.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.12.1.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.12.0.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.11.3.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.11.2.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.11.1.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.11.0.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.10.3.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.10.2.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.10.1.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.10.0.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.9.1.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.9.0.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.8.1.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.8.0.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.7.2.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.7.1.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.7.0.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.6.8.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.6.7.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.6.6.tar.gz (1.7 MB)\n",
      "  Downloading uvloop-0.6.5.tar.gz (2.0 MB)\n",
      "  Downloading uvloop-0.6.0.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.5.5.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.5.4.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.5.3.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.5.2.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.5.1.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.5.0.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.34.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.33.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.32.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.31.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.30.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.29.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.28.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.27.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.26.tar.gz (1.9 MB)\n",
      "  Downloading uvloop-0.4.25.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.24.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.23.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.22.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.21.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.20.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.19.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.18.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.17.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.16.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.15.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.14.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.13.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.12.tar.gz (1.8 MB)\n",
      "  Downloading uvloop-0.4.11.tar.gz (845 kB)\n",
      "  Downloading uvloop-0.4.10.tar.gz (845 kB)\n",
      "  Downloading uvloop-0.4.9.tar.gz (844 kB)\n",
      "  Downloading uvloop-0.4.8.tar.gz (844 kB)\n",
      "  Downloading uvloop-0.4.7.tar.gz (840 kB)\n",
      "  Downloading uvloop-0.4.6.tar.gz (790 kB)\n",
      "  Downloading uvloop-0.4.5.tar.gz (778 kB)\n",
      "  Downloading uvloop-0.4.4.tar.gz (1.7 MB)\n",
      "Building wheels for collected packages: uvloop\n",
      "  Building wheel for uvloop (setup.py): started\n",
      "  Building wheel for uvloop (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for uvloop\n",
      "Failed to build uvloop\n",
      "Installing collected packages: uvloop\n",
      "    Running setup.py install for uvloop: started\n",
      "    Running setup.py install for uvloop: finished with status 'error'\n",
      "Collecting cchardet\n",
      "  Downloading cchardet-2.1.7-cp39-cp39-win_amd64.whl (115 kB)\n",
      "Installing collected packages: cchardet\n",
      "Successfully installed cchardet-2.1.7\n",
      "Collecting aiodns\n",
      "  Downloading aiodns-3.0.0-py3-none-any.whl (5.0 kB)\n",
      "Collecting pycares>=4.0.0\n",
      "  Downloading pycares-4.1.2-cp39-cp39-win_amd64.whl (83 kB)\n",
      "Requirement already satisfied: cffi>=1.5.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from pycares>=4.0.0->aiodns) (1.14.6)\n",
      "Requirement already satisfied: pycparser in c:\\programdata\\anaconda3\\lib\\site-packages (from cffi>=1.5.0->pycares>=4.0.0->aiodns) (2.20)\n",
      "Installing collected packages: pycares, aiodns\n",
      "Successfully installed aiodns-3.0.0 pycares-4.1.2\n",
      "Collecting aiohttp[speedups]\n",
      "  Downloading aiohttp-3.8.1-cp39-cp39-win_amd64.whl (554 kB)\n",
      "Collecting frozenlist>=1.1.1\n",
      "  Downloading frozenlist-1.3.0-cp39-cp39-win_amd64.whl (33 kB)\n",
      "Collecting async-timeout<5.0,>=4.0.0a3\n",
      "  Downloading async_timeout-4.0.2-py3-none-any.whl (5.8 kB)\n",
      "Collecting multidict<7.0,>=4.5\n",
      "  Downloading multidict-6.0.2-cp39-cp39-win_amd64.whl (28 kB)\n",
      "Requirement already satisfied: attrs>=17.3.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp[speedups]) (21.2.0)\n",
      "Requirement already satisfied: charset-normalizer<3.0,>=2.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp[speedups]) (2.0.4)\n",
      "Collecting yarl<2.0,>=1.0\n",
      "  Downloading yarl-1.7.2-cp39-cp39-win_amd64.whl (122 kB)\n",
      "Collecting aiosignal>=1.1.2\n",
      "  Downloading aiosignal-1.2.0-py3-none-any.whl (8.2 kB)\n",
      "Collecting Brotli\n",
      "  Downloading Brotli-1.0.9-cp39-cp39-win_amd64.whl (383 kB)\n",
      "Requirement already satisfied: aiodns in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp[speedups]) (3.0.0)\n",
      "Requirement already satisfied: cchardet in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp[speedups]) (2.1.7)\n",
      "Requirement already satisfied: idna>=2.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from yarl<2.0,>=1.0->aiohttp[speedups]) (3.2)\n",
      "Requirement already satisfied: pycares>=4.0.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiodns->aiohttp[speedups]) (4.1.2)\n",
      "Requirement already satisfied: cffi>=1.5.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from pycares>=4.0.0->aiodns->aiohttp[speedups]) (1.14.6)\n",
      "Requirement already satisfied: pycparser in c:\\programdata\\anaconda3\\lib\\site-packages (from cffi>=1.5.0->pycares>=4.0.0->aiodns->aiohttp[speedups]) (2.20)\n",
      "Installing collected packages: multidict, frozenlist, yarl, async-timeout, aiosignal, Brotli, aiohttp\n",
      "Successfully installed Brotli-1.0.9 aiohttp-3.8.1 aiosignal-1.2.0 async-timeout-4.0.2 frozenlist-1.3.0 multidict-6.0.2 yarl-1.7.2\n",
      "Collecting aiogram\n",
      "  Downloading aiogram-2.20-py3-none-any.whl (192 kB)\n",
      "Requirement already satisfied: certifi>=2021.10.8 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiogram) (2021.10.8)\n",
      "Requirement already satisfied: Babel<2.10.0,>=2.9.1 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiogram) (2.9.1)\n",
      "Requirement already satisfied: aiohttp<3.9.0,>=3.8.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiogram) (3.8.1)\n",
      "Requirement already satisfied: frozenlist>=1.1.1 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (1.3.0)\n",
      "Requirement already satisfied: yarl<2.0,>=1.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (1.7.2)\n",
      "Requirement already satisfied: multidict<7.0,>=4.5 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (6.0.2)\n",
      "Requirement already satisfied: async-timeout<5.0,>=4.0.0a3 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (4.0.2)\n",
      "Requirement already satisfied: aiosignal>=1.1.2 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (1.2.0)\n",
      "Requirement already satisfied: attrs>=17.3.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (21.2.0)\n",
      "Requirement already satisfied: charset-normalizer<3.0,>=2.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from aiohttp<3.9.0,>=3.8.0->aiogram) (2.0.4)\n",
      "Requirement already satisfied: pytz>=2015.7 in c:\\programdata\\anaconda3\\lib\\site-packages (from Babel<2.10.0,>=2.9.1->aiogram) (2021.3)\n",
      "Requirement already satisfied: idna>=2.0 in c:\\programdata\\anaconda3\\lib\\site-packages (from yarl<2.0,>=1.0->aiohttp<3.9.0,>=3.8.0->aiogram) (3.2)\n",
      "Installing collected packages: aiogram\n",
      "Successfully installed aiogram-2.20\n"
     ]
    }
   ],
   "source": [
    "! pip install ujson\n",
    "! pip install uvloop\n",
    "! pip install cchardet\n",
    "! pip install aiodns\n",
    "! pip install aiohttp[speedups]\n",
    "! pip install aiogram"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "99cae976",
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "from aiogram import Bot, Dispatcher, executor, types\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ddb15c78",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}